usermods/audioreactive/audio_source.h

```diff
@@ -490,75 +490,139 @@ class ES7243 : public I2SSource {
     int8_t pin_ES7243_SCL;
 };
 
+/* ES8388 Microphone
+   This is an I2S microphone that requires ininitialization over
+   I2C before I2S data can be received
+*/
 class ES8388Source : public I2SSource {
   private:
-
-    void _es8388InitAdc() {
-      DEBUGSR_PRINTF("\nAR: _es8388InitAdc\n"); 
-
-      if (!es8388.init()) {
-        Serial.println("_es8388InitAdc init Fail\n");
-        return;
+    // I2C initialization functions for ES8388
+    void _es8388I2cBegin() {
+      bool i2c_initialized = Wire.begin(pin_ES8388_SDA, pin_ES8388_SCL, 100000U);
+      if (i2c_initialized == false) {
+        ERRORSR_PRINTLN(F("AR: ES8388 failed to initialize I2C bus driver."));
       }
-      es8388.inputSelect(IN2); // IN2  Line-In
-      es8388.setInputGain(8);
-      es8388.outputSelect(OUT1);  // OUT1 - Headphones
-      es8388.setOutputVolume(12);
-      es8388.mixerSourceSelect(MIXADC, MIXADC);
-      es8388.mixerSourceControl(DACOUT);
-      es8388.setALCmode(MUSIC);
-      es8388.analogBypass(true); 
-      uint8_t *reg;
-      for (uint8_t i = 0; i < 53; i++) {
-        reg = es8388.readAllReg();
-        DEBUGSR_PRINTF("Reg-%02d = 0x%02x\r\n", i, reg[i]);
+    }
+
+    void _es8388I2cWrite(uint8_t reg, uint8_t val) {
+#ifndef ES8388_ADDR
+      Wire.beginTransmission(0x10);
+      #define ES8388_ADDR 0x10   // default address
+#else
+      Wire.beginTransmission(ES7243_ADDR);
+#endif
+      Wire.write((uint8_t)reg);
+      Wire.write((uint8_t)val);
+      uint8_t i2cErr = Wire.endTransmission();  // i2cErr == 0 means OK
+      if (i2cErr != 0) {
+        DEBUGSR_PRINTF("AR: ES8388 I2C write failed with error=%d  (addr=0x%X, reg 0x%X, val 0x%X).\n", i2cErr, ES8388_ADDR, reg, val);
       }
     }
 
-public:
-    ES8388Source(int sampleRate, int blockSize, int sdaPin, int sclPin) :
-      I2SSource(sampleRate, blockSize) {
+    void _es8388InitAdc() {
+      // This is by no means 100% figured but it's working for line-in
+      // with a little too much noise for my liking...
+      _es8388I2cBegin();
+      _es8388I2cWrite(0x08,0x00);
```

Comment:
This section could really benefit for some comments, maybe not every value, but just which group of values are which, e.g input selection, i2s etc

Reply:
Added and optimized. Still working on this in general to see if I can't get the line noise down.